presto-common/src/main/java/com/facebook/presto/common/type/DoubleType.java

```diff
@@ -149,7 +153,11 @@ public final BlockBuilder createFixedSizeBlockBuilder(int positionCount)
     @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
     public boolean equals(Object other)
     {
-        return other == DOUBLE;
+        // this is safe because we will only have one type of double in the system at
+        // a time. .equals() comparison is always used against the static DOUBLE
+        // instance to check if something is double type. this hack is temporary
+        // and will be removed when we full remove the old nan behavior
+        return other == DOUBLE || other == OLD_NAN_DOUBLE;
```

Comment:
Should we just do an instanceof check?  It wouldn't require a lengthy explanation.  Likewise for float.